### PR TITLE
Track trading times for user and liquidity orders separately

### PR DESCRIPTION
**Pull request by MartinquaXD**
The introduction of more liquidity orders made the "Average Trade Time" panel less meaningful since we don't really care about taking forever to match liquidity orders.
Now we only update those metrics with settled user orders to better gauge what the actual UX is like.

### Test Plan
CI

### CODE REVIEW

**Review by sunce86 — APPROVED** [review]
Makes sense.

File: crates/solver/src/driver.rs [hunk 1]
```diff
@@ -170,7 +170,12 @@ impl Driver {
         rated_settlement: RatedSettlement,
     ) -> Result<TransactionReceipt> {
         let settlement = rated_settlement.settlement;
-        let traded_orders = settlement.traded_orders().cloned().collect::<Vec<_>>();
+        let user_orders = settlement
```
Comment (nlordell) [inline crates/solver/src/driver.rs:173]: Ah, OK. So the solution is easy.

**Review by nlordell — CHANGES_REQUESTED** [review]
Nice! Happy there is a simple solution.

That being said, I would request a change where we have metrics for both user orders and liquidity orders (so we can keep track of how many those trade as well). Since we pass in the full order to `order_settled` we can just differentiate on `is_liquidity_order`.

**Comment by MartinquaXD** [comment]
> Since we pass in the full order to order_settled we can just differentiate on is_liquidity_order.

It's not as easy as that. Only `LimitOrder` has the field `is_liquidity_order` but we are passing in `Order`s. Since you already suggested to store `is_liquidity_order` in the `OrderMetadata` maybe we should start with that and put this PR on hold until then? But that change would also affect the database.
Otherwise I can easily get the information whether an order is a liquidity order elsewhere and track their metrics separately.

**Comment by nlordell** [comment]
> maybe we should start with that and put this PR

This would be my suggestion. The other advantages of this are:
- You know which are liquidity orders (which have different semantics for objective value computation than regular orders) when retrieving the orderbook.
- You don't need to duplicate the `--liquidity-order-owner` configuration as we do now.
- The "liquidity order status" of an order correctly survives across restarts.

Not sure how others feel about this though.

**Comment by vkgnosis** [comment]
It makes sense to me. If we do this we should come up with a good textual description of what it means to be a liquidity order since that will be part of the public api.

**Comment by MartinquaXD** [comment]
I'm all for it. 👍

File: crates/solver/src/driver.rs [hunk 2]
```diff
@@ -164,14 +166,39 @@ impl Driver {
...
+            let mut group = group.into_iter().peekable();
+            let order = &group.peek().unwrap().0.order;
+            let previously_filled = match order.creation.kind {
+                OrderKind::Buy => order.metadata.executed_buy_amount.clone(),
+                OrderKind::Sell => order.metadata.executed_sell_amount.clone(),
+            };
+            let newly_filled = group.fold(BigUint::default(), |acc, (trade, _)| {
+                acc + u256_to_big_uint(&trade.executed_amount)
+            });
+            if previously_filled == 0u8.into() && newly_filled >= 0u8.into() {
```
Comment (nlordell) [inline crates/solver/src/driver.rs:187]: Technically, a solver could fill `0`, meaning we might count orders more than once. We should likely:
```suggestion
            if previously_filled == 0u8.into() && newly_filled > 0u8.into() {
```

**Review by nlordell — APPROVED** [review]
LGTM.

I think there is one tiny edge case. Approving as either I am mistaken about the edge case or assume it will be resolved.

File: crates/solver/src/driver.rs (hunk 2 above)
Comment (MartinquaXD) [reply crates/solver/src/driver.rs:187]: 🤦‍♂️ 
Good catch!

**Pull request merged**